### PR TITLE
Application/Controller: store component parameters in context (#2007)

### PR DESCRIPTION
**Pull request by neinkeinkaffee**
**What this PR does / why we need it**: 

Currently the options for passing context information about a component to an attached trait are limited when the component is defined in Terraform/HCL.

This PR is partial quick-fix, not a fundamental solution to the problem. It effects that the `properties` configured for a given component in the `ApplicationDefinition` get stored in `context.parameters`, so that the values are accessible to a given component's traits.

**Which issue(s) this PR fixes**: #2007

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]
LGTM, thanks

File: pkg/cue/process/handle.go [hunk 1]
```diff
@@ -139,6 +146,11 @@ func (ctx *templateContext) SetConfigs(configs []map[string]string) {
 	ctx.configs = configs
 }
 
+// SetParameters sets templateContext parameters
+func (ctx *templateContext) SetParameters(params map[string]interface{}) {
+	ctx.parameters = params
+}
```
Comment (wonderflow) [inline pkg/cue/process/handle.go:152]: can you help add tests to cover these lines

Comment (neinkeinkaffee) [reply pkg/cue/process/handle.go:152]: Ah, interesting! I would have to think of a way. I just wonder why they're not covered by the fact that SetParameters is invoked in [pkg/cue/process/handle_test.go](https://github.com/neinkeinkaffee/kubevela/blob/d4bbd139dfbedf25990536191673900c5c7698d0/pkg/cue/process/handle_test.go#L83)...?

Comment (neinkeinkaffee) [reply pkg/cue/process/handle.go:152]: I can't see another test where the corresponding methods of the Context interface are covered, apart from handle_test.go. Could you suggest a good place where I might place an additional test for `SetParameters`?

**Comment by leejanee** [comment]
good job

File: pkg/cue/process/handle.go (hunk 1 above)
Comment (wonderflow) [reply pkg/cue/process/handle.go:152]: Oh, then it's ok, let me merge it

**Pull request merged**